helix-term/src/ui/statusline.rs

```diff
@@ -432,12 +432,14 @@ fn render_file_modification_indicator<F>(context: &mut RenderContext, write: F)
 where
     F: Fn(&mut RenderContext, String, Option<Style>) + Copy,
 {
-    let title = (if context.doc.is_modified() {
+    debug_assert!(!(context.doc.readonly.unwrap_or(false) && context.doc.is_modified()));
+    let title = (if context.doc.readonly.unwrap_or(false) {
+        "[readonly]"
+    } else if context.doc.is_modified() {
```

Comment:
Thus should be an else (see abovw)

Reply:
Should there be a warning message if a user is modified a readonly file? Vim has: `Warning: Changing a readonly file`

Reply:
Yeah that what I mean with:

>  Ideally, we would have more extensive read-only handling, see https://github.com/helix-editor/helix/pull/7128#issuecomment-1581560237 although a status line indicator that indicates the file is not writable might be a reasonable first step.

But that would be a larger chabge. You can work in that jf you want ti but it would require some design discussion (look at what kakune does)  and would require more substantial changes si it might be better to keep that for future work

Reply:
I was thinking along the lines of this:
```
let title = match (context.doc.readonly, context.doc.is_modified()) {
    (false, false) => "   ",
    (false, true) => "[+]",
    (true, false) => "[readonly]",
    (true, true) => "Warning: Changing a readonly file",
    }.to_string();
```
If it's too ugly then I can leave it out but ugly is probably better here since the user probably wants to know this.